README.md

```diff
@@ -98,142 +113,140 @@ oc login -u developer -p developer https://localhost:8443
+### Authentication and Authorization
 
-An official support for Apache Spark on OpenShift is provided by the [radanalytics.io](https://radanalytics.io/) project by means of
-the Oshinko application with a Web UI for deploying a Spark cluster. Other than using such a Web UI, a CLI tool is available as well which is used for this workshop.
+In this workshop we aim to setup a secure-by-default IoT solution, so we need to define the 
+applications and what addresses they need to access. Before we create the bindings we need, lets
+define the mapping:
 
-Go to [Oshinko CLI downloads](https://github.com/radanalyticsio/oshinko-cli/releases) and download
-the latest release (0.3.1 as of time of writing). Unpack the release:
+* deviceX - send: temperature, recv: control/deviceX
+* spark-driver - recv: temperature, send: max
+* thermostat - recv: max, send: control*
 
-```
-tar xvf oshinko_v0.3.1_linux_amd64.tar.gz
-```
-
-Then use the following command in order to deploy a Spark cluster made by one master node and one slave node.
-
-```
-export SPARK_NAME=<something>
-
-./oshinko_linux_amd64/oshinko create $SPARK_NAME --masters=1 --workers=1
-```
+We will create the bindings to each of the application as we deploy them.
 
 ### Deploying the "Temperature Analyzer" Spark driver
 
-The `spark-driver` directory provides the Spark Streaming driver application and a Docker image for running the related Spark driver inside the cluster. Using `minishift`, we need that the following Docker images (built from the source) will be available in the local `minishift` Docker registry. For this reason, it's needed to run the following command first :
+The `spark-driver` directory provides the Spark Streaming driver application and a Docker image for running the related Spark driver inside the cluster. The spark-driver is deployed by building and running it on the OpenShift cluster.  The spark-driver uses the [fabric8-maven-plugin](https://github.com/fabric8io/fabric8-maven-plugin) to create a docker image, an OpenShift deployment config, and deploy the spark-driver into OpenShift.
+
+To deploy the spark driver:
 
 ```
-eval $(minishift docker-env)
+mvn clean package fabric8:resource fabric8:build fabric8:deploy -Dspark.master.host=myspark.spark.svc -Dspark.app=myapp
 ```
 
-In order to set as `DOCKER_HOST`, the `minishift` environment and all the Docker images will be built using that environment and will be available in `minishift` Docker registry.
+This command will package the application and build a Docker image deployed to OpenShift.
 
-This application can be packaged running the following command from such directory.
+Once the driver has been deployed, we need to create a binding with the permissions we defined above. Click on "Create binding" to open the dialog to create a binding. Set `sendAddresses` to `max` and `recvAddresses` to `temperature`.
 
-```
-mvn clean install package -Pbuild-docker-image
-```
+![Binding1](images/binding1.png)
```

Comment:
I cannot see this image in the images folder.

Reply:
fixed